File: frontend/src/shared/api/types/catalog.ts
export type ProductPublic = {
  productId: number
  ticketTypeId: number
  scenicSpotName: string
  productName: string
  ticketName: string
  ticketCategory: string
  originalPrice: string
  salePrice: string
  description?: string | null
  refundRule?: string | null
  realNameRequired: boolean
  tripType: string
  raftCapacity: number
  startPierName: string
  endPierName: string
  windowPhone: string
}

export type TimeSlotPublic = {
  timeSlotId: number
  productId: number
  ticketTypeId: number
  visitDate: string
  slotStartTime: string
  slotEndTime: string
  quotaRemaining: number
}

File: backend/app/schemas/catalog.py
from datetime import date, time
from decimal import Decimal

from pydantic import BaseModel, Field


class ProductPublicDTO(BaseModel):
    product_id: int = Field(alias="productId")
    ticket_type_id: int = Field(alias="ticketTypeId")
    scenic_spot_name: str = Field(alias="scenicSpotName")
    product_name: str = Field(alias="productName")
    ticket_name: str = Field(alias="ticketName")
    ticket_category: str = Field(alias="ticketCategory")
    original_price: Decimal = Field(alias="originalPrice")
    sale_price: Decimal = Field(alias="salePrice")
    description: str | None = None
    refund_rule: str | None = Field(default=None, alias="refundRule")
    real_name_required: bool = Field(alias="realNameRequired")
    trip_type: str = Field(alias="tripType")
    raft_capacity: int = Field(alias="raftCapacity")
    start_pier_name: str = Field(alias="startPierName")
    end_pier_name: str = Field(alias="endPierName")
    window_phone: str = Field(alias="windowPhone")

    model_config = {"populate_by_name": True}


class TimeSlotPublicDTO(BaseModel):
    time_slot_id: int = Field(alias="timeSlotId")
    product_id: int = Field(alias="productId")
    ticket_type_id: int = Field(alias="ticketTypeId")
    visit_date: date = Field(alias="visitDate")
    slot_start_time: time = Field(alias="slotStartTime")
    slot_end_time: time = Field(alias="slotEndTime")
    quota_remaining: int = Field(alias="quotaRemaining")

    model_config = {"populate_by_name": True}
